src/Forms/TextField.php

```diff
@@ -83,6 +111,11 @@ public function getSchemaDataDefaults()
     {
         $data = parent::getSchemaDataDefaults();
         $data['data']['maxlength'] =  $this->getMaxLength();
+
+        if ($this->tip instanceof Tip) {
```

Comment:
Generally, I prefer using the getter/setter rather than accessing private properties. That way there's never any doubt about  bypassing some validation rules or some magical accessor/mutator logic.

But that's just a personal preference.

Reply:
Couple of ideas. 
- We can interface this (eg. `TippedFieldInterface`) that declares a `getTip` method? The `setTip` probably doesn't need to belong in the interface. Then you can just do `if ($this instanceof TippedFieldInterface` here.
- We can probably extract the implementation of `getTip` and `setTip` to a trait to save on code reuse. Then we can maybe move this logic to `FormField`?

Reply:
The trait could also provide a helper method for adding a tip too - `->addTip(string $tip): Tip`?

Reply:
I've added a `TippableFieldInterface` which declares both `getTip(): ?Tip` and `setTip(?Tip $tip)`, as I'd rather see a consistent method of both getting and setting the tip, and `addTip` would suggest there could be multiple.

It still makes sense for this conditional to be checking against what `$this->getTip()` is returning, as it could be `null`, and if so we shouldn't try to use it. The interface might be helpful for actually determining whether this feature is available when a module developer tries to use it:

```php
if ($myTextField instanceof TippableFieldInterface) {
    $myTextField->setTip(...);
}
```

I think the trait will make sense if/when we spread this functionality to other fields, but I don't think it's necessary to create until then, as we can cleanly shift the methods / property out when it's appropriate (and it may not be useful at all if we end up shifting the logic up to `FormField`).

Reply:
The only reason I recommended against having `setTip` in the interface as it seems like an unnecessary restriction on the implementor. What if I create a `InspriationalQuoteTippedField implements TippableFieldInterface` that creates a new `Tip` with an inspirational quote every time you call `getTip`?

Reply:
I think it makes more sense for the interface to indicate how the object is used rather than how it might be configured - or at least those can be in different interfaces (but that's sounding a bit like interface overload at this point).

Reply:
😂 I appreciate the out-of-box thinking, but ultimately the goal of this feature is to allow someone to add their own extended description to a field. Having it in the interface means that the 'feature flag' example I provided will actually guarantee that the `setTip` call will work, even if it's ultimately a no-op for `InspirationalQuoteTippedField`.

That's also why I changed the name from `Tipped` to `Tippable`  - the former designates that it is always 'tipped', while the latter designates that it _can be_ 'tipped'.